solutions/swb-reference/src/services/sshKeyService.test.ts

```diff
@@ -85,29 +91,160 @@ describe('SshKeyService', () => {
+          beforeEach(() => {
+            const keyPairs = [{ Tags: [{ Key: 'user', Value: `${deleteSshKeyRequest.currentUserId}` }] }];
+            hostEc2.describeKeyPairs = jest.fn(() => Promise.resolve({ $metadata: {}, KeyPairs: keyPairs }));
+          });
+
+          describe('but no key exists', () => {
+            beforeEach(() => {
+              hostEc2.describeKeyPairs = jest.fn(() => Promise.resolve({ $metadata: {}, KeyPairs: [] }));
+            });
+
+            test('it throws NoKeyExistsError', async () => {
+              // OPERATE n CHECK
+              await expect(() => sshKeyService.deleteSshKey(deleteSshKeyRequest)).rejects.toThrow(
+                NoKeyExistsError
+              );
+            });
+          });
+
+          describe('but multiple keys exists', () => {
+            beforeEach(() => {
+              const keyPairs = [
+                { Tags: [{ Key: 'user', Value: `${deleteSshKeyRequest.currentUserId}` }] },
+                { Tags: [{ Key: 'user', Value: `${deleteSshKeyRequest.currentUserId}` }] }
+              ];
+              hostEc2.describeKeyPairs = jest.fn(() =>
+                Promise.resolve({ $metadata: {}, KeyPairs: keyPairs })
+              );
+            });
+
+            test('it throws NoKeyExistsError', async () => {
+              // OPERATE n CHECK
+              await expect(() => sshKeyService.deleteSshKey(deleteSshKeyRequest)).rejects.toThrow(
+                NonUniqueKeyError
+              );
+            });
+          });
+
+          describe('but current user does not own the key they want to delete', () => {
+            beforeEach(() => {
+              deleteSshKeyRequest.currentUserId = 'user-456';
+            });
+
+            test('it throws ForbiddenError', async () => {
+              // OPERATE n CHECK
+              await expect(() => sshKeyService.deleteSshKey(deleteSshKeyRequest)).rejects.toThrow(
+                ForbiddenError
+              );
+            });
+          });
+
+          describe('and current user owns the existing, unique key they want to delete', () => {
+            describe('but delete EC2 call fails', () => {
+              beforeEach(() => {
+                hostEc2.deleteKeyPair = jest.fn(() => Promise.reject('Some EC2 thrown error'));
+              });
+
+              test('it throws Ec2Error', async () => {
+                // OPERATE n CHECK
+                await expect(() => sshKeyService.deleteSshKey(deleteSshKeyRequest)).rejects.toThrow(Ec2Error);
+              });
+            });
+
+            describe('and get EC2 call succeeds', () => {
+              beforeEach(() => {
+                hostEc2.deleteKeyPair = jest.fn(() => Promise.resolve({ $metadata: {} }));
+              });
+              test('it succeeds, nothing is returned', async () => {
+                // OPERATE n CHECK
+                await expect(sshKeyService.deleteSshKey(deleteSshKeyRequest)).resolves.not.toThrow();
```

Comment:
Do you want to check the deleteSshKey method actually delete the key (call describeKeyPairs again), or are you planning to done that in integration test

Reply:
This is a mocked call to `describeKeyPairs` and `deleteKeyPair`. The check you are referring to will be a multi-step integ test!

Reply:
got it!